layer2/ObjectMap.cpp

```diff
@@ -5956,3 +5957,56 @@ ObjectMap *ObjectMapLoadChemPyMap(PyMOLGlobals * G, ObjectMap * I, PyObject * Ma
   return (I);
 #endif
 }
+
+void ObjectMapDump(ObjectMap* om, const char* fname, int state, int quiet)
+{
+  auto file = fopen(fname, "wb");
+  if (!file) {
+    ErrMessage(om->G, "ObjectMapDump", "can't open file for writing");
+  }
+  
+  ObjectMapState* oms = &om->State[state];
+  
+  int xd = oms->Field->dimensions[0];
+  int yd = oms->Field->dimensions[1];
+  int zd = oms->Field->dimensions[2];
+  
+  float xs = (oms->ExtentMax[0] - oms->ExtentMin[0]) / (oms->Field->dimensions[0]);
+  float ys = (oms->ExtentMax[1] - oms->ExtentMin[1]) / (oms->Field->dimensions[1]);
+  float zs = (oms->ExtentMax[2] - oms->ExtentMin[2]) / (oms->Field->dimensions[2]);
+  
+  int offset = 0;
+  for (int xi = 0; xi < oms->Field->dimensions[0]; xi++) {
+    for (int yi = 0; yi < oms->Field->dimensions[1]; yi++) {
+      for (int zi = 0; zi < oms->Field->dimensions[2]; zi++) {
+        float x = oms->ExtentMin[0] + xs * xi;
+        float y = oms->ExtentMin[1] + ys * yi;
+        float z = oms->ExtentMin[2] + zs * zi;
```

Comment:
I think the `(ExtentMax - ExtentMin) / dimensions` calculation is off by one grid point? Maybe it makes sense to use `Isofield::points` instead.
```c++
  const CField *field = oms->Field->data;
  const CField *points = oms->Field->points;

  for (int xi = 0; xi < oms->Field->dimensions[0]; xi++) {
    for (int yi = 0; yi < oms->Field->dimensions[1]; yi++) {
      for (int zi = 0; zi < oms->Field->dimensions[2]; zi++) {
        float x = F4Ptr(points, xi, yi, zi, 0);
        float y = F4Ptr(points, xi, yi, zi, 1);
        float z = F4Ptr(points, xi, yi, zi, 2);
```